blas/impl/KokkosBlas1_nrm1_impl.hpp

```diff
@@ -142,33 +139,33 @@ void MV_Nrm1_Invoke(
   }
   // Zero out the result vector
   Kokkos::deep_copy(
-      execution_space(), r,
-      Kokkos::ArithTraits<typename RV::non_const_value_type>::zero());
+      space, r, Kokkos::ArithTraits<typename RV::non_const_value_type>::zero());
   size_type teamsPerVec;
   KokkosBlas::Impl::multipleReductionWorkDistribution<execution_space,
                                                       size_type>(
       x.extent(0), x.extent(1), teamsPerVec);
   size_type numTeams = x.extent(1) * teamsPerVec;
-  Kokkos::TeamPolicy<execution_space> pol(numTeams, Kokkos::AUTO);
+  Kokkos::TeamPolicy<execution_space> pol(space, numTeams, Kokkos::AUTO);
   Kokkos::parallel_for(
       "KokkosBlas1::Nrm1::S1", pol,
       Nrm1_MV_Functor<execution_space, RV, XV, size_type>(r, x, teamsPerVec));
 }
 
 // Version for when a temporary result view is needed (implemented in terms of
 // the other version)
-template <class RV, class XV, class size_type>
+template <class execution_space, class RV, class XV, class size_type>
 void MV_Nrm1_Invoke(
-    const RV& r, const XV& x,
+    const execution_space& space, const RV& r, const XV& x,
     typename std::enable_if<!Kokkos::SpaceAccessibility<
-        typename XV::execution_space,
-        typename RV::memory_space>::accessible>::type* = nullptr) {
+        execution_space, typename RV::memory_space>::accessible>::type* =
+        nullptr) {
   Kokkos::View<typename RV::non_const_value_type*, typename XV::memory_space>
       tempResult(
           Kokkos::view_alloc(Kokkos::WithoutInitializing, "Nrm1 temp result"),
           r.extent(0));
-  MV_Nrm1_Invoke<decltype(tempResult), XV, size_type>(tempResult, x);
-  Kokkos::deep_copy(typename XV::execution_space(), r, tempResult);
+  MV_Nrm1_Invoke<execution_space, decltype(tempResult), XV, size_type>(
+      space, tempResult, x);
+  Kokkos::deep_copy(space, r, tempResult);
```

Comment:
Uh oh, this version has had a bug whole time.  ``tempResult`` goes out of scope and gets deallocated while the deep_copy could still be running. Since there's no way to check if an async deep_copy has finished, I can't think of a way to fix this other than ``space.fence();`` after the copy. It's still asynchronous with respect to all other streams, at least.